lib/game.rb

```diff
@@ -0,0 +1,119 @@
+class Game
+  attr_accessor :player_1, :player_2, :board
+
+  WIN_COMBINATIONS = [
+    [0, 1, 2],
+    [3, 4, 5],
+    [6, 7, 8],
+    [0, 3, 6],
+    [1, 4, 7],
+    [2, 5, 8],
+    [0, 4, 8],
+    [2, 4, 6]
+  ]
+
+  def initialize(player_1=Players::Human.new("X"), player_2=Players::Human.new("O"), board=Board.new)
+    @player_1 = player_1 if player_1
+    @player_2 = player_2 if player_2
+    @board = board if board
+    self.play
+  end
+
+  def current_player
+    x_count = self.board.cells.count {|cell| cell == "X"}
+    o_count = self.board.cells.count {|cell| cell == "O"}
+    if x_count > o_count
+      return player_2
+    else
+      return player_1
+    end
+  end
+
+  def won?
+    WIN_COMBINATIONS.each do |win_combination|
+      if ((self.board.cells[win_combination[0]]) == (self.board.cells[win_combination[1]])) &&
+        ((self.board.cells[win_combination[1]]) == (self.board.cells[win_combination[2]])) &&
+        ((self.board.cells[win_combination[0]]) != " ") &&
+        ((self.board.cells[win_combination[1]]) != " ") &&
+        ((self.board.cells[win_combination[2]]) != " ")
+        return true
+      end
+    end
+    return false
+  end
+
+  def draw?
+    self.board.full? && !self.won?
+  end
+
+  def winner
```

Comment:
You can utilize the `won?` method to determine the winning token. 
```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.